### ISSUE TITLE
"RecursionError: maximum recursion depth exceeded in comparison"

### ISSUE THREAD

**Issue by cwmoriarty** [jprouty/mint-amazon-tagger#122]
Seeing the following on a long-term search:

```Getting Mint Transactions|More transactions are available than max page size - try reducing the date range.

Parsing Mint Transactions |████████████████████████████████| 10000/10000
Matching Amazon Items with Orders |███████████████████████████████▌| 5220/5295Traceback (most recent call last):
  File "/usr/local/bin/mint-amazon-tagger-cli", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/cli.py", line 118, in main
    results = tagger.create_updates(
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/tagger.py", line 154, in create_updates
    updates, unmatched_orders = get_mint_updates(
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/tagger.py", line 240, in get_mint_updates
    amazon.associate_items_with_orders(orders, items, itemProgress)
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/amazon.py", line 201, in associate_items_with_orders
    for item_groupings in algorithm_u(oid_items, len(orders)):
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/algorithm_u.py", line 20, in f
    for v in f(mu - 1, nu - 1, (mu + sigma) % 2, n, a):
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/algorithm_u.py", line 20, in f
    for v in f(mu - 1, nu - 1, (mu + sigma) % 2, n, a):
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/algorithm_u.py", line 20, in f
    for v in f(mu - 1, nu - 1, (mu + sigma) % 2, n, a):
  [Previous line repeated 991 more times]
  File "/usr/local/lib/python3.10/site-packages/mintamazontagger/algorithm_u.py", line 17, in f
    if mu == 2:
RecursionError: maximum recursion depth exceeded in comparison
```

**Comment by jprouty** [comment]
Ugh; sorry you ran into this! Tagger is trying to brute force determine which items from an order when into which shipment. I'll add some logic to catch this exception, as well as a timeout to prevent this from taking too long (I believe you have an old bug report something to this effect - might have been this same routine.)

**Comment by jprouty** [comment]
A fix to this issue will be in 1.61. Sorry for the delay, as I imagine this error was likely giving you repeated trouble (requiring manual effort to exclude the problematic order).
